examples/src/test/scala/akka/projection/cassandra/scaladsl/WordCountDocExampleSpec.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.projection.cassandra.scaladsl
+
+import java.util.UUID
+
+import scala.compat.java8.FutureConverters._
+import scala.concurrent.Await
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.Duration
+import scala.concurrent.duration._
+import scala.util.Try
+
+import akka.Done
+import akka.actor.testkit.typed.scaladsl.LogCapturing
+import akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit
+import akka.projection.Projection
+import akka.projection.ProjectionId
+import akka.projection.cassandra.ContainerSessionProvider
+import akka.projection.cassandra.internal.CassandraOffsetStore
+import akka.projection.testkit.scaladsl.ProjectionTestKit
+import akka.stream.alpakka.cassandra.scaladsl.CassandraSessionRegistry
+import docs.cassandra.WordCountDocExample._
+import org.scalatest.wordspec.AnyWordSpecLike
+
+class WordCountDocExampleSpec
+    extends ScalaTestWithActorTestKit(ContainerSessionProvider.Config)
+    with AnyWordSpecLike
+    with LogCapturing {
+
+  private val session = CassandraSessionRegistry(system).sessionFor("akka.projection.cassandra")
+  private implicit val ec: ExecutionContext = system.executionContext
+  private val offsetStore = new CassandraOffsetStore(session)
+  private val repository = new CassandraWordCountRepository(session)
+  private val projectionTestKit = new ProjectionTestKit(testKit)
+
+  override protected def beforeAll(): Unit = {
+    super.beforeAll()
+
+    // don't use futureValue (patience) here because it can take a while to start the test container
+    Await.result(ContainerSessionProvider.started, 30.seconds)
+
+    Await.result(for {
+      session <- session.underlying()
+      // reason for setSchemaMetadataEnabled is that it speed up tests
+      _ <- session.setSchemaMetadataEnabled(false).toScala
+      // FIXME move this test to docs.cassandra package when CassandraOffsetStore.createKeyspaceAndTable is public
+      _ <- offsetStore.createKeyspaceAndTable()
+      _ <- repository.createKeyspaceAndTable()
+      _ <- session.setSchemaMetadataEnabled(null).toScala
+    } yield Done, 30.seconds)
+  }
+
+  override protected def afterAll(): Unit = {
+    // reason for setSchemaMetadataEnabled is that it speed up tests
+    Try(session.underlying().map(_.setSchemaMetadataEnabled(false)).futureValue)
+    Try(session.executeDDL(s"DROP keyspace ${offsetStore.keyspace}").futureValue)
+    Try(session.executeDDL(s"DROP keyspace ${repository.keyspace}").futureValue)
+    Try(session.underlying().map(_.setSchemaMetadataEnabled(null)).futureValue)
```

Comment:
note to self: hm, copied from another test, I thought I changed this to for comprehension